### ISSUE TITLE
map.from_unique_list

### ISSUE THREAD

**Issue by CrowdHailer** [gleam-lang/suggestions#74]
A function that will turn a list to a map or return an error if a duplicate key is found. 

```
from_unique_list: fn(List(tuple(key, value))) -> Result(Map(key, value), key) 
```

Returned error value is the first key

**Comment by lpil** [comment]
This isn't a function I've used before- is there precedence in other languages' standard libraries? What's the use case?

**Comment by CrowdHailer** [comment]
The usecase is in http query strings being able to show a user an error for parameters that shouldn't be declared twice but where

**Comment by lpil** [comment]
I've had a look around but I can't find any existing languages with this function- do you know of any?

**Comment by CrowdHailer** [comment]
No, but I my experience most applications don't actually show duplicate keys as a warning. so at some level it is missing. Quite happy for this to be something that lives in my parameters library for a while

**Comment by lpil** [comment]
Let's keep it there for now, and we can promote it if it turns out to be commonly useful.
